src/containers/app/container.tsx

```diff
@@ -0,0 +1,105 @@
+import {
+  always, append, identity, is, mergeDeepRight,
+  evolve, cond, not, propEq, filter, map, T, reject
+} from 'ramda';
+import * as React from 'react';
+
+import { container } from 'architecture';
+import Message, { Activate } from 'architecture/message';
+
+import { TodoFilter, TodoAppModel } from './model';
+import App from './index';
+
+class AddTodo extends Message {};
+
+class ShowFilter extends Message {};
+
+class ClearCompleted extends Message {};
+
+class DeleteTodo extends Message {
+  public static expects = { value: is(Number) };
+};
+
+class CompleteTodo extends Message {
+  public static expects = { value: is(Number) };
+};
+
+class EditTodo extends Message {
+  public static expects = { value: is(Number) };
+};
+
+const createTodoId = () => new Date().valueOf();
+
+const emptyInput = {
+  text: '',
+};
+
+export default container({
```

Comment:
You should be able to do `export default container<TodoAppModel>({` here and then not have to re-type the model in your updaters.